headertest/dummy_header.go

```diff
@@ -100,6 +105,10 @@ func (d *DummyHeader) IsExpired(period time.Duration) bool {
 }
 
 func (d *DummyHeader) Verify(header header.Header) error {
+	if dummy, _ := header.(*DummyHeader); dummy.VerifyFailure {
```

Comment:
If `header` is not a `*DummyHeader` this will panic, right? Is this intended?

Reply:
Yes, intended. It would be strange (even in a test) to have this occur unless that's the case that's explicitly being tested for.